lib/umwelt/files.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Umwelt.Files do
+  @moduledoc "Finds elixir files recursive from dir"
+
+  def list_from_root(project),
+    do: files_in("lib/#{project}")
+
+  def files_in(path) do
+    {:ok, files} = File.ls(path)
+
+    files
+    |> Enum.map(&extract("#{path}/#{&1}"))
+    |> List.flatten()
+  end
+
+  defp extract(path) do
+    if File.dir?(path) do
+      files_in(path)
+    else
+      path
+    end
+  end
```

Comment:
I hate `if/2` for no reason.
```suggestion
  defp extract(path), do: path |> File.dir?() |> do_extract(path)

  defp do_extract(true, path), do: files_in(path)
  defp do_extract(false, path), do: path
```

That way you immediately see this whole function is redundant and might be moved to `files_in/1` as

```elixir
  def files_in(path), do: path |> File.dir?() |> do_files_in(path)

  defp do_files_in(false, path), do: path
  defp do_files_in(true, path) do
    with {:ok, files} <- File.ls(path),
      do: Enum.flat_map(files, &files_in(Path.join(path, &1)))
  end
```

Reply:
I hate `if` too, but from time to time I start implement something and can forget that it can be done via matching.